### PR TITLE
Unsetting data in case produced data is null

### PR DESCRIPTION
**Pull request by r0goyal**

### CODE REVIEW

File: src/test/java/com/flipkart/databuilderframework/NullableDataTest.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        DataDelta dataDelta = new DataDelta(Lists.newArrayList(new A()));
+        DataExecutionResponse response = executor.run(dataFlowInstance, dataDelta);
+        Assert.assertEquals(1, response.getResponses().size());
+
+        A a = new A();
+        a.setRetry(true);
+
+        dataDelta = new DataDelta(Lists.<Data>newArrayList(a));
+        response = executor.run(dataFlowInstance, dataDelta);
+        Assert.assertEquals(0, response.getResponses().size());
```
Comment (santanusinha) [inline src/test/java/com/flipkart/databuilderframework/NullableDataTest.java]: Assert data is missing from current dataset

File: src/test/java/com/flipkart/databuilderframework/NullableDataTest.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            val accessor = context.getDataSet().accessor();
+            val retryableData = accessor.get(A.class);
+            if (retryableData.isRetry()) {
+                return null;
+            }
+            return new B();
+        }
+
+    }
+
```
Comment (santanusinha) [inline src/test/java/com/flipkart/databuilderframework/NullableDataTest.java]: Add one more level to make sure it's not getting triggered in retry case.

File: src/test/java/com/flipkart/databuilderframework/NullableDataTest.java (hunk 1 above)
Comment (r0goyal) [reply src/test/java/com/flipkart/databuilderframework/NullableDataTest.java]: Dataset is already empty.

Comment (santanusinha) [reply src/test/java/com/flipkart/databuilderframework/NullableDataTest.java]: missed it

**Pull request merged**